### ISSUE TITLE
Can't remove magic areas?

### ISSUE THREAD

**Issue by jscherry** [jseidl/hass-magic_areas#116]

**Comment by jscherry** [comment]
I’m trying to remove the magic areas. Uninstalled from hacs. Deleted all the areas entities. Deleted all the magic areas in integrations and the magic area integration disapeared but comes back always after reboot.Removed magic areas from yaml. It still shows up under integrations every reboot and I get this error on start up
Invalid config

The following integrations and platforms could not be set up:

magic_areas
Please check your config and logs.

and this error in logs:

Logger: homeassistant.setup
Source: setup.py:172
First occurred: 10:37:29 PM (1 occurrences)
Last logged: 10:37:29 PM

Setup failed for magic_areas: Integration not found.

What am I missing?

**Comment by jseidl** [comment]
Thanks for submitting this report. It should be straightforward, what you did should have removed it. 

Can you check in the integrations page if it still shows up? I need to do some testing on this

**Comment by jscherry** [comment]
It comes back every time under integrations with all the areas and no devices? I’ve triple checked all my config files and have removed everything I could find associated with magic areas. The first time I removed it I had to delete the individual areas under integrations then the integration went away but immediately showed back up under integrations after reboot. Now there’s no option to remove the integration because each area has no devices. I’ve removed the hacs integration? It’s not hurting anything it just drives me nuts that I can’t figure out how to get rid of it. I’m rather knowledgeable about home assistant and have been using it for over 3 years but I’ve never ran into something like this I couldn’t delete, remove, or uninstall? The first time I tried to get rid of it I deleted the hacs integration first, then I went to integrations and deleted individual areas, the integration disappeared, rebooted and it came back so I checked entities and saw the areas so I removed those?
Thanks 
Jeff

**Comment by jscherry** [comment]
Did you ever find a solution for removing it? All the areas still show up under the main integration page. When you click on an area it just says not loaded and the 3 dots that would allow you to delete it are missing?

**Comment by Lockszmith** [comment]
@jscherry, I agree with you - it should be easier to remove, but at first I got stuck like you.

Here is what worked for me:
1. Make sure Magic Areas integration is working:
    a. Add the Magic Areas in HACS
    b. make sure it's in your configuration.yaml.
    c. Restart HA core.
2. In integration - remove each of the magic areas (yes, this needs to be done one by one). For each magic-area:
    a. Select the magic area
    b. Click the options (three dots) and choose Delete.
3. When all have been removed - remove entry from configuration.yaml
4. I also removed the integration from my HACS instance, but I think this is optional
5. Restart HA core.

After all of these steps, Magic Area was completely removed.

To @jseidl I would add, If the integration is manageable form the UI, it should probably be possible to add and completely removed from the UI.
I think the mixture of YAML configuration with UI based integration makes things more difficult than they should be.

**Comment by jseidl** [comment]
I agree with you and apologize for the hassle. 

There are some technical reasons why I can't make it work fully through the UI. As MagicAreas scans the Area Registry and creates the area entries at runtime, it doesn't work through the UI flow, only to edit the autocreated entries.

I'm quite new to making components and also been quite busy at work so that's why its taking a long time but I definitely have this as a priority on the to-do list

**Comment by Lockszmith** [comment]
Hey @jseidl, no need to apologize, you're already doing something others don't _AND_ you are sharing it with the world.

Keep it up! and thanks for making this.
